stl/inc/format

```diff
@@ -719,27 +717,23 @@ public:
                   using _Td = remove_const_t<_Ty>;
                   // doesn't drop const-qualifier per an unnumbered LWG issue
                   using _Tq = conditional_t<_Formattable_with<const _Ty, _Context>, const _Ty, _Ty>;
-                  if constexpr (_Formattable_with_non_const<_Tq, _Context>) {
-                      static_assert(_Formattable_with<_Tq, _Context>,
-                          "The format() member function can't be called on const formatter<T>. "
-                          "To make the formatter usable, add const to format(). "
-                          "See N4971 [format.arg]/12, [format.formattable], and [formatter.requirements].");
-                  } else {
-                      static_assert(_Formattable_with<_Tq, _Context>,
-                          "Cannot format an argument. "
-                          "To make this type formattable, provide a formatter<T> specialization. "
-                          "See N4971 [format.arg]/12, [format.formattable], and [formatter.requirements].");
-                  }
+                  static_assert(_Formattable_with<_Tq, _Context>);

```

Comment:
If we believe that user code should never be able to trigger this without having already caused errors elsewhere, such that we don't need to emit an explanatory message, then this should be an `_STL_INTERNAL_STATIC_ASSERT`.

Reply:
The `static_assert` used to implemented [[format.arg]/12](https://eel.is/c++draft/format.arg#12) (originally introduced by WG21-P2418R2 and modified by LWG-3631). I believe the `static_assert`s in `make_format_args` (originally added in #3080 in order to check _Preconditions_ in [[format.arg.store]/2](https://eel.is/c++draft/format.arg.store#2)) already covers [format.arg]/12.

Although it's a bit curious to me to show `static_assert` messages indicating _Preconditions_ instead of _Mandates_...

Reply:
Emitting `static_assert`s for violations of *Preconditions* is a "nice to have"/quality of implementation thing - if it's not too verbose and can improve the user diagnostic experience then I'm generally in favor of it.